nixos/modules/services/networking/bee.nix

```diff
@@ -0,0 +1,166 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.bee;
+in {
+  meta = {
+    # doc = ./bee.xml;
+    maintainers = with maintainers; [ attila-lendvai ];
+  };
+
+  ### interface
+
+  options = {
+    services.bee = {
+      enable = mkEnableOption "Start up Ethereum Swarm Bee";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.bee;
+        defaultText = "pkgs.bee";
+        example = "pkgs.bee-unstable";
+        description = "The package providing the bee binary for the service.";
+      };
+
+      ### NOTE let's call the nix variables the same as bee calls them
+
+      config = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/var/lib/bee/bee.yaml";
+        #description = "The configuration file path.";
+      };
+
+      api-addr = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''HTTP API listen address (default ":1633")'';
+      };
+
+      data-dir = mkOption {
+        type = types.path;
+        default = "/var/lib/bee/";
+        #description = "The data directory.";
+      };
+
+      extraConfig = mkOption {
```

Comment:
With the above suggestion of adding a `settings` option this option would become obsolete.